client/modules/IDE/pages/IDEView.jsx

```diff
@@ -78,347 +79,256 @@ function WarnIfUnsavedChanges() {
 
 export const CmControllerContext = React.createContext({});
 
-class IDEView extends React.Component {
-  constructor(props) {
-    super(props);
-    this.handleGlobalKeydown = this.handleGlobalKeydown.bind(this);
-
-    this.state = {
-      consoleSize: props.ide.consoleIsExpanded ? 150 : 29,
-      sidebarSize: props.ide.sidebarIsExpanded ? 160 : 20
-    };
-  }
+const IDEView = (props) => {
+  const ide = useSelector((state) => state.ide);
+  const selectedFile = useSelector(selectActiveFile);
+  const preferences = useSelector((state) => state.preferences);
+  const project = useSelector((state) => state.project);
+  const isUserOwner = useSelector(getIsUserOwner);
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
 
-  componentDidMount() {
-    // If page doesn't reload after Sign In then we need
-    // to force cleared state to be cleared
-    this.props.clearPersistedState();
+  const [consoleSize, setConsoleSize] = useState(
+    ide.consoleIsExpanded ? 150 : 29
+  );
+  const [sidebarSize, setSidebarSize] = useState(
+    ide.sidebarIsExpanded ? 160 : 20
```

Comment:
A lot of my feedback here is going to be stuff that we already discussed on Discord.

You can simplify the handling of the size state by storing just the expanded size.  So `useState(160)`

Reply:
Okay